arch/riscv/core/isr.S

```diff
@@ -323,17 +423,23 @@ skip_store_fp_caller_saved:
 
 #ifdef CONFIG_RISCV_SOC_CONTEXT_SAVE
 	/* Handle context saving at SOC level. */
+	addi sp, sp, -16
+	RV_OP_STOREREG a0, 0x00(sp)
+
 	addi a0, sp, __z_arch_esf_t_soc_context_OFFSET
```

Comment:
In my opinion, $a0 to __soc_save_context is incorrect since $sp has been updated, soc context may pollute caller save context.